### PR TITLE
PR feedback for https://github.com/aspnet/KestrelHttpServer/pull/2838

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Following up for comments in previous PR. Not merging until release/2.2 is open.

### CODE REVIEW

File: src/Kestrel.Core/Http2Limits.cs [hunk 1]
```diff
@@ -62,17 +58,17 @@ public int HeaderTableSize
         /// <summary>
         /// Indicates the size of the largest frame payload that is allowed to be received, in octets. The size must be between 2^14 and 2^24-1.
         /// <para>
-        /// Defaults to 2^14 (16,384)
+        ///Value must be between 2^14 and 2^24, defaults to 2^14 (16,384)
```
Comment (Tratcher) [inline src/Kestrel.Core/Http2Limits.cs]: Space

File: src/Kestrel.Core/Internal/Http2/Http2PeerSettings.cs [hunk 2]
```diff
@@ -7,15 +7,20 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2
...
         public const uint DefaultHeaderTableSize = 4096;
         public const bool DefaultEnablePush = true;
         public const uint DefaultMaxConcurrentStreams = uint.MaxValue;
         public const uint DefaultInitialWindowSize = 65535;
-        public const uint DefaultMaxFrameSize = Http2Limits.MinAllowedMaxFrameSize;
+        public const uint DefaultMaxFrameSize = MinAllowedMaxFrameSize;
         public const uint DefaultMaxHeaderListSize = uint.MaxValue;
         public const uint MaxWindowSize = int.MaxValue;
 
+        // These are limits defined by the RFC https://tools.ietf.org/html/rfc7540#section-4.2, not Kestrel defaults.
```
Comment (Tratcher) [inline src/Kestrel.Core/Internal/Http2/Http2PeerSettings.cs]: This comment is redundant with the one above. Remove it and the empty line above?

File: test/Kestrel.InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs [hunk 3]
```diff
@@ -77,32 +77,32 @@ public class Http2ConnectionTests : Http2TestBase
...
         [Fact]
         public async Task ServerSettings_ChangesRequestMaxFrameSize()
         {
-            var length = Http2Limits.MinAllowedMaxFrameSize + 10;
+            var length = Http2PeerSettings.MinAllowedMaxFrameSize + 10;
             _connectionContext.ServiceContext.ServerOptions.Limits.Http2.MaxFrameSize = length;
             _connection = new Http2Connection(_connectionContext);
 
-            await InitializeConnectionAsync(_echoApplication, expectedSettingsLegnth: 12);
+            await InitializeConnectionAsync(_echoApplication, expectedSettingsCount: 2);
```
Comment (Tratcher) [inline test/Kestrel.InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs]: I've already changed this in my PR. You can skip it to avoid the conflict.

**Comment by JunTaoLuo** [comment]
:up::date:

**Pull request merged**